core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -267,44 +264,20 @@ private[spark] object Utils extends Logging {
     }
   }
 
-  /**
-   * JDK equivalent of `chmod 700 file`.
-   *
-   * @param file the file whose permissions will be modified
-   * @return true if the permissions were successfully changed, false otherwise.
-   */
-  def chmod700(file: File): Boolean = {
-    file.setReadable(false, false) &&
-    file.setReadable(true, true) &&
-    file.setWritable(false, false) &&
-    file.setWritable(true, true) &&
-    file.setExecutable(false, false) &&
-    file.setExecutable(true, true)
-  }
-
   /**
    * Create a directory inside the given parent directory. The directory is guaranteed to be
    * newly created, and is not marked for automatic deletion.
    */
   def createDirectory(root: String, namePrefix: String = "spark"): File = {
-    var attempts = 0
-    val maxAttempts = MAX_DIR_CREATION_ATTEMPTS
-    var dir: File = null
-    while (dir == null) {
-      attempts += 1
-      if (attempts > maxAttempts) {
-        throw new IOException("Failed to create a temp directory (under " + root + ") after " +
-          maxAttempts + " attempts!")
-      }
-      try {
-        dir = new File(root, namePrefix + "-" + UUID.randomUUID.toString)
-        if (dir.exists() || !dir.mkdirs()) {
-          dir = null
-        }
-      } catch { case e: SecurityException => dir = null; }
+    val prefix = namePrefix + "-"
+    val dir = if (root != null) {
+      val rootDir = new File(root)
+      require(rootDir.isDirectory(), s"Root path $root must be an existing directory.")
```

Comment:
Why not use the nio api's? e.g.:
```scala
val rootDir = Paths.get(root)
require(Files.isDirectory(rootDir), s"Root path $root must be an existing directory.")
Files.createTempDirectory(rootDir, prefix)
```

Reply:
Same thing in the end, no? This method needs to return `File` regardless of what it does internally.

Reply:
That is also an argument for not changing this method at all. Just use `File.createTempFile(...)` and be done with it.

Reply:
This is for a directory, not file. `File.createTempFile` does not work.

Reply:
Yeah, I missed that sorry. However my point still stands.

Reply:
Not sure what that point really is, but if you feel so strongly about using the nio APIs, that's easy enough.